ratchet_core/src/handshake/mod.rs

```diff
@@ -245,11 +228,59 @@ impl<'l, 'h, 'buf: 'h> TryMap<Request> for &'l httparse::Request<'h, 'buf> {
                 )))
             }
         };
+        let method = match self.method {
+            Some(m) => {
+                Method::from_str(m).map_err(|_| HttpError::HttpMethod(Some(m.to_string())))?
+            }
+            None => return Err(HttpError::HttpMethod(None)),
+        };
+        let version = match self.version {
+            Some(v) => match v {
+                0 => Version::HTTP_10,
+                1 => Version::HTTP_11,
+                n => return Err(HttpError::HttpVersion(Some(n))),
+            },
+            None => return Err(HttpError::HttpVersion(None)),
+        };
         let headers = &self.headers;
 
         *request.headers_mut() = headers.try_map()?;
         *request.uri_mut() = path;
+        *request.version_mut() = version;
+        *request.method_mut() = method;
 
         Ok(request)
     }
 }
+
+impl<'l, 'h, 'buf: 'h> TryMap<Response<()>> for &'l httparse::Response<'h, 'buf> {
```

Comment:
This can be simplified to `impl<'a, 'b> TryMap<Response<()>> for &'a httparse::Response<'b, 'b>`.

Also, I'd probably create some sort of `Wrapper<'a T>' the imlement the standard library trait for it rather than writing your own trait (with a somewhat confusing name).